modules/proto_tls/proto_tls.c

```diff
@@ -618,7 +618,7 @@ static int proto_tls_send(const struct socket_info* send_sock,
 	send_sock->last_real_ports->local = c->rcv.dst_port;
 	send_sock->last_real_ports->remote = c->rcv.src_port;
 
-	tcp_conn_release(c, 0);
+	tcp_conn_release(c, c->async && c->async->pending?1:0);
```

Comment:
OK, let's keep it local, here. Nevertheless, to avoid any clarity or synchronization issue (btw, the access to `->async->pending` should be under lock), I would go for the more simpler approach from the [proto_tcp](https://github.com/OpenSIPS/opensips/blob/master/net/proto_tcp/proto_tcp.c#L588), with the test on the amount of written data. Could you :
* rework the test here
* remove the `tcp_connections_no` fix (to keep it in  #3664)